### PR TITLE
chore: Update lavamoat to a version with more diff-friendly policy ordering

### PR DESCRIPTION
**Pull request by naugtur**
## **Description**

This lavamoat update brings a different sorting comparator for policy.json files that will produce more readable diffs.
This PR has 2 commits - one that reorders the policy without making any changes and another that updates lavamoat packages on top of that.

For better clarity on the policy.json files, inspect each commit separately.

This is going to be hard to review and merge because conflict resolution requires a redo. Gonna have to schedule it carefully.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/29311?quickstart=1)

## **Manual testing steps**

1. run a local policy update and verify the diff is not ridiculously large (that would mean the update failed to apply properly after policy got reordered here)

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by naugtur** [comment]
@metamaskbot update-policies

File: lavamoat/browserify/main/policy.json [hunk 1]
```diff
@@ -2693,7 +2693,7 @@
         "@zxing/library>ts-custom-error": true
       }
     },
-    "extension-port-stream>readable-stream>abort-controller": {
+    "@lavamoat/lavapack>readable-stream>abort-controller": {
```
Comment (naugtur) [inline lavamoat/browserify/main/policy.json:2696]: None of these browserify policies should have changed, so let's explore why these changes are trivial:
- the path to abort-controller changed because packages got rearranged in the tree and this is now the shortest path to it.
- readable-stream got deduplicated into a newer version that uses AbortSignal directly etc.

File: lavamoat/build-system/policy.json [hunk 2]
```diff
@@ -1225,14 +1171,21 @@
         "assert": true,
         "path": true
       },
+      "globals": {
+        "afterAll": true,
+        "process.cwd": true
```
Comment (naugtur) [inline lavamoat/build-system/policy.json:1176]: This seems like we're detecting new globals we didn't detect before

File: lavamoat/build-system/policy.json [hunk 3]
```diff
@@ -2976,6 +2921,13 @@
         "debounce-stream>through": true
       }
     },
+    "@lavamoat/lavapack>readable-stream>abort-controller>event-target-shim": {
+      "globals": {
+        "Event": true,
+        "EventTarget": true,
+        "console": true
```
Comment (naugtur) [inline lavamoat/build-system/policy.json:2928]: We could restrict these in overrides later as I don't think this shim has any work left to do.

**Comment by naugtur** [comment]
Policy review:
There should be no meningful changes to policy in the webapp as we didn't touch anything there. the changes present are due to changes in the shape of the dependency tree.

Changes to the build policy indicate the lavamoat update (the readable-stream is a new copy of readable-stream (lavamoat updated to v4 and it's the newest version in the dependencies here according to yarn why)

cc @MetaMask/policy-reviewers

**Review by Mrtenz — DISMISSED** [review]
Looks good!

**Review by weizman — DISMISSED** [review]
Lg

**Comment by naugtur** [comment]
> Bundle size diffs [🚨 Warning! Bundle size has increased!]
same exact commits on main 3d ago said decreased :shrug:

**Comment by naugtur** [comment]
@metamaskbot update-policies

**Review by Mrtenz — APPROVED** [review]

**Review by danroc — APPROVED** [review]
